src/Tizen.Network.Stc/Tizen.Network.Stc/StcEnumerations.cs

```diff
@@ -0,0 +1,258 @@
+﻿/*
+ * Copyright (c) 2016 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using Tizen.Internals.Errors;
+
+namespace Tizen.Network.Stc
+{
+    /// <summary>
+    /// Enumeration for Stc time period.
+    /// </summary>
+    public enum StcTimePeriod
+    {
+        /// <summary>
+        /// Unknown
+        /// </summary>
+        Unknown = 0,
```

Comment:
Consider removing `Unknown` value. 

Why need `Unknown` value? 
it seem to a error case if get `Unknown` value. 
If it is really error case, you should throw exception instead of `Unknown` value

Reply:
This is as per CAPI Enums.

Reply:
It is not CAPI. it can't be reason
if CAPI returing Unknown you should throw exception

Reply:
I am uncertain about how to do it. Because,
1) If "Interop.Stc.Rule.GetTimePeriod(_ruleHandle, out timePeriod)" gave **'Unknown'** value for CAPI enum 'stc_time_period_e', then it needs to be handled by corresponding C# enum 'StcTimePeriod'.

2) If somehow 'StcTimePeriod' enum variable does not get initialized or assigned a value from the getter method, then it has a default value of zero. And that's why we need to declare a zero(Unknown) inside this enum to avoid errors.
Reference: "https://blog.submain.com/c-enum-definition-examples-best-practices-and-pitfalls/" says: "Pitfall #2: Assign a value to zero to avoid business logic errors in switch statements."

3) I think that getters don't throw exceptions.
Reference: "https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property"
says: "X AVOID throwing exceptions from property getters."

Reply:
>     1. If "Interop.Stc.Rule.GetTimePeriod(_ruleHandle, out timePeriod)" gave **'Unknown'** value for CAPI enum 'stc_time_period_e', then it needs to be handled by corresponding C# enum 'StcTimePeriod'.
First, CAPI will return `stc_time_period_e` `Unknown` only error cases, 
So, You should report error to app developer. 

>     2\. If somehow 'StcTimePeriod' enum variable does not get initialized or assigned a value from the getter method, then it has a default value of zero. And that's why we need to declare a zero(Unknown) inside this enum to avoid errors.
        Reference: "https://blog.submain.com/c-enum-definition-examples-best-practices-and-pitfalls/" says: "Pitfall #2: Assign a value to zero to avoid business logic errors in switch statements."
Yes, You should not assign a zero value(like Unknown) to avoid error case

>    3\. I think that getters don't throw exceptions.
        Reference: "https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property"
        says: "X AVOID throwing exceptions from property getters."

Yes, It is **AVOID** not **DO NOT**
And, you can chage to method not property

Reply:
I think, we can ignore a error when only a default value is working well on error case
for example, 
about On/Off enum values, Off value is working well as default value on error case
But your case not working as default value
``` c#
StcRule stcRule1 = ...
StcRule stcRule2 = ...
var period = stcRule1.TimePeriod;
...
stcRule2.TimePeriod = period; // It will cause exception
```

Reply:
Shall i do like this in the getter?
->
                 if(timePeriod == TimePeriodType.Unknown)
                 {
                     Log.Error(Globals.LogTag, "Getting 'Unknown' as timePeriod");
                     throw new InvalidOperationException("Operation failed");
                 }
                 else
                     return timePeriod;

Reply:
No, You should revmoe Unknown value in TimePeriodType enum

``` c#
   if (nativePeriod == NativeTimePeriodType.Unknow)
          throw new InvalidOperationException( ... );
   else
          return (TimePeriodType)nativePeriod;
```

Reply:
Done for all similar enums.